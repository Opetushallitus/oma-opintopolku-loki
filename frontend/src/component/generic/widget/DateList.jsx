import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'
import { take } from 'ramda'

const MARGIN = '1rem'

const DateTable = styled.table`
  padding: 0 ${MARGIN} ${MARGIN} ${MARGIN};
`

const DateList = ({ dates, numShown }) => {
  const subList = numShown ? take(numShown, dates) : dates

  return (
    <DateTable>
      <tbody>
        {subList.map((d, i) => (
          <tr key={`${d}_${i}`}>
            <td>{d}</td>
          </tr>
        ))}
      </tbody>
    </DateTable>
  )
}

DateList.propTypes = {
  dates: PropTypes.array.isRequired,
  numShown: PropTypes.number
}

export default DateList
