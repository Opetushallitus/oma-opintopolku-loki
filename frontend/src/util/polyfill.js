import 'core-js/features/promise'
import 'core-js/features/string/starts-with'
