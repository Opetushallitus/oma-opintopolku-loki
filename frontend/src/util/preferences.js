import Cookies from 'js-cookie'

export const lang = Cookies.get('lang') || 'fi'
