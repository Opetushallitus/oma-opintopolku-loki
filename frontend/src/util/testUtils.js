/* istanbul ignore file */
import renderer from 'react-test-renderer'

export function matchesSnapshot (component) {
  const renderedComponent = renderer.create(component)
  const componentJSON = renderedComponent.toJSON()
  expect(componentJSON).toMatchSnapshot()
  renderedComponent.unmount()
}
