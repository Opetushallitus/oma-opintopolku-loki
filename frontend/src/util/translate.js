import Cookies from 'js-cookie'

export default key => {
  const lang = Cookies.get('lang') || 'fi'
  const translations = window.translationsMap || {}
  const versions = translations[key] || {}
  const translated = versions[lang]

  if (!translated) {
    console.error(`Translation missing for language ${lang}: ${key}`)
    return key
  }

  return translated
}
