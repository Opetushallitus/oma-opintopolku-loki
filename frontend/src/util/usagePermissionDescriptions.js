import t from 'util/translate'

import permissionsMap from 'Resources/mapping/usagePermissionDescriptions'

export const getTranslatedUsagePermissionDescription = organizationOid => {
  const permission = permissionsMap[organizationOid] || permissionsMap.default
  return t(permission)
}

export const isMydataPartner = organizationOid => {
  return Boolean(permissionsMap[organizationOid])
}
