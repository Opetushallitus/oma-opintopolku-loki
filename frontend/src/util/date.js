import moment from 'moment'
import { lang } from 'util/preferences'

moment.locale(lang)

export const isoStringToDate = isoString => moment(isoString).format('l, LT')
