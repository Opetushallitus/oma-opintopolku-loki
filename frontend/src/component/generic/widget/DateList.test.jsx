import React from 'react'
import DateList from './DateList'
import { matchesSnapshot } from 'util/testUtils'

describe('DateList', () => {
  // eslint-disable-next-line jest/expect-expect
  it('should be rendered', () => {
    matchesSnapshot(<DateList dates={['1.1.2018']}/>)
  })
})
