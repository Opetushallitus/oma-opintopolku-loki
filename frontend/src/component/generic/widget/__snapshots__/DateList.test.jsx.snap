// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DateList should be rendered 1`] = `
<table
  className="DateList__DateTable-sc-9hn0em-0 kUpUuj"
>
  <tbody>
    <tr>
      <td>
        1.1.2018
      </td>
    </tr>
  </tbody>
</table>
`;
